asyncpg/transaction.py

```diff
@@ -89,50 +103,59 @@ async def __aexit__(self, extype, ex, tb):
     @connresource.guarded
     async def start(self):
         """Enter the transaction or savepoint block."""
-        self.__check_state_base('start')
+        self.__check_state_base("start")
         if self._state is TransactionState.STARTED:
             raise apg_errors.InterfaceError(
-                'cannot start; the transaction is already started')
+                "cannot start; the transaction is already started"
+            )
 
         con = self._connection
 
         if con._top_xact is None:
             if con._protocol.is_in_transaction():
+                logger.error(
```

Comment:
Make structlog logging.error call with the connection's id and recent statements list before raising the transaction state-confused error.